### PR TITLE
zipkin: Fix HTTP_JSON uint64 serialization

### LINKED ISSUE #9341: Zipkin HTTP_JSON v2 spans using strings for timestamp/duration
*Title*: Zipkin HTTP_JSON v2 payload using strings for timestamp/duration

*Description*:
According to the Zipkin v2 API (https://zipkin.io/zipkin-api/#/default/post_spans) spans should contain `timestamp` and `duration` fields as `int64`, not strings, in the JSON payload. This makes collectors not being able to parse the JSON payload.

When configuring Zipkin to use the `HTTP_JSON` endpoint version, the resulting payload is:

```
[
  {
    "traceId": "83713138fa14c726",
    "id": "83713138fa14c726",
    "kind": "SERVER",
    "name": "www.hello",
    "timestamp": "1572849988488551",  <-------------------------------------------
    "duration": "7131",        <-------------------------------------------
    "localEndpoint": {
      "serviceName": "envoy-cluster",
      "ipv4": "172.31.0.5"
    },
    "tags": {
      "response_flags": "-",
      "request_size": "0",
      "response_size": "292",
      "node_id": "envoy",
      "http.status_code": "200",
      "http.method": "GET",
      "upstream_cluster": "testcluster",
      "guid:x-request-id": "3ab0fc56-f92e-9c03-a087-1f23662699cb",
      "zone": "us-east-1",
      "http.protocol": "HTTP/1.1",
      "component": "proxy",
      "http.url": "https://www.hello/headers",
      "downstream_cluster": "-",
      "user_agent": "HTTPie/0.9.2"
    }
  }
]
```
When it should be

```
[
  {
    "traceId": "83713138fa14c726",
    "id": "83713138fa14c726",
    "kind": "SERVER",
    "name": "www.hello",
    "timestamp": 1572849988488551,  <-------------------------------------------
    "duration": 7131,        <-------------------------------------------
    "localEndpoint": {
      "serviceName": "envoy-cluster",
      "ipv4": "172.31.0.5"
    },
    "tags": {
      "response_flags": "-",
      "request_size": "0",
      "response_size": "292",
      "node_id": "envoy",
      "http.status_code": "200",
      "http.method": "GET",
      "upstream_cluster": "testcluster",
      "guid:x-request-id": "3ab0fc56-f92e-9c03-a087-1f23662699cb",
      "zone": "us-east-1",
      "http.protocol": "HTTP/1.1",
      "component": "proxy",
      "http.url": "https://www.hello/headers",
      "downstream_cluster": "-",
      "user_agent": "HTTPie/0.9.2"
    }
  }
]
```

*Repro steps*:
Enable Zipkin tracing with `HTTP_JSON` as follows:

```
tracing:
  http:
    name: envoy.zipkin
    config:
      collector_cluster: tracing-sidecar
      collector_endpoint: /api/v2/spans
      collector_endpoint_version: HTTP_JSON
      shared_span_context: false
```

### PR DESCRIPTION
**Pull request by dio**
This patch fixes the serialization of uint64 fields to be numbers instead of strings. This also removes rapidjson from the envoy zipkin extension code. 

Risk Level: Low
Testing: Unit tests and manual tests using in-repo Zipkin and Jaeger examples.
Docs Changes: N/A
Release Notes: N/A
Fixes #9341

Signed-off-by: Dhi Aurrahman <dio@tetrate.io>

### CODE REVIEW

**Comment by moderation** [comment]
It might be interesting trying to bump the protobuf dependency as part of this PR considering the JSON to protobuf changes. I tried in the past but protobuf 3.11.2 failed to build based on failures in the Zipkin code.

To try use the following in `bazel/repository_locations.bzl:
```bazel
        sha256 = "b679cef31102ed8beddc39ecfd6368ee311cbee6f50742f13f21be7278781821",
        strip_prefix = "protobuf-3.11.2",
        urls = ["https://github.com/protocolbuffers/protobuf/releases/download/v3.11.2/protobuf-all-3.11.2.tar.gz"],
```

**Comment by dio** [comment]
@moderation from my observation, directly upgrading like above introduces compilation due to enforced `[-Werror,-Wdeprecated-declarations]`? e.g. 

```
bazel-out/k8-fastbuild/bin/external/envoy_api/envoy/api/v2/core/config_source.pb.h:106:73: error: 'ApiConfigSource_ApiType_UNSUPPORTED_REST_LEGACY' is deprecated [-Werror,-Wdeprecated-declarations]
constexpr ApiConfigSource_ApiType ApiConfigSource_ApiType_ApiType_MIN = ApiConfigSource_ApiType_UNSUPPORTED_REST_LEGACY;
                                                                        ^
bazel-out/k8-fastbuild/bin/external/envoy_api/envoy/api/v2/core/config_source.pb.h:98:51: note: 'ApiConfigSource_ApiType_UNSUPPORTED_REST_LEGACY' has been explicitly marked deprecated here
  ApiConfigSource_ApiType_UNSUPPORTED_REST_LEGACY PROTOBUF_DEPRECATED_ENUM = 0,
                                                  ^
external/com_google_protobuf/src/google/protobuf/port_def.inc:152:49: note: expanded from macro 'PROTOBUF_DEPRECATED_ENUM'
#define PROTOBUF_DEPRECATED_ENUM __attribute__((deprecated))
```

**Comment by moderation** [comment]
@dio This is a change that needs to happen in the core Zipkin code or does the error come from somewhere else?

File: source/common/protobuf/utility.cc [hunk 1]
```diff
@@ -582,6 +582,32 @@ bool ValueUtil::equal(const ProtobufWkt::Value& v1, const ProtobufWkt::Value& v2
   }
 }
 
+ProtobufWkt::Value ValueUtil::stringValue(const std::string& str) {
```
Comment (zuercher) [inline source/common/protobuf/utility.cc]: Do you mind dropping a todo into the access log formatter code to switch to these helpers (or if you want extra karma points, convert it to use these)?

Comment (dio) [reply source/common/protobuf/utility.cc]: Sure!

Comment (dio) [reply source/common/protobuf/utility.cc]: https://github.com/envoyproxy/envoy/pull/9432/commits/4598ff29573e714883a83e98801ff205bbe49655

File: source/common/access_log/access_log_formatter.cc [hunk 2]
```diff
@@ -24,30 +25,11 @@ namespace Envoy {
 namespace AccessLog {
 
 static const std::string UnspecifiedValueString = "-";
-static const std::string EmptyString = "";
+static const std::string EmptyString;
```
Comment (lizan) [inline source/common/access_log/access_log_formatter.cc]: Use source/common/common/empty_string.h

Comment (dio) [reply source/common/access_log/access_log_formatter.cc]: Done

File: source/extensions/tracers/zipkin/span_buffer.cc [hunk 3]
```diff
@@ -73,76 +73,89 @@ JsonV2Serializer::JsonV2Serializer(const bool shared_span_context)
...
     }
 
-    auto& tags = *span.mutable_tags();
-    for (const auto& binary_annotation : zipkin_span.binaryAnnotations()) {
-      tags[binary_annotation.key()] = binary_annotation.value();
+    const auto& binary_annotations = zipkin_span.binaryAnnotations();
+    if (!binary_annotations.empty()) {
+      ProtobufWkt::Struct tags;
+      for (const auto& binary_annotation : binary_annotations) {
+        auto* tag_fields = tags.mutable_fields();
```
Comment (zuercher) [inline source/extensions/tracers/zipkin/span_buffer.cc]: nit: move this outside the loop?

File: test/common/access_log/access_log_formatter_test.cc [hunk 4]
```diff
@@ -106,7 +107,7 @@ TEST(AccessLogFormatterTest, plainStringFormatter) {
 
   EXPECT_EQ("plain", formatter.format(header, header, header, stream_info));
   EXPECT_THAT(formatter.formatValue(header, header, header, stream_info),
-              ProtoEq(stringValue("plain")));
+              ProtoEq(ValueUtil::stringValue("plain")));
```
Comment (zuercher) [inline test/common/access_log/access_log_formatter_test.cc]: Near the top of this file (e.g. lines 33-52 or so), there are test-private implementations of these new ValueUtil functions that can be deleted.

File: source/extensions/tracers/zipkin/span_buffer.cc (hunk 3 above)
Comment (dio) [reply source/extensions/tracers/zipkin/span_buffer.cc]: 726f7e3

File: test/common/access_log/access_log_formatter_test.cc (hunk 4 above)
Comment (dio) [reply test/common/access_log/access_log_formatter_test.cc]: 726f7e3

**Pull request merged**